hack/get-kubeconfig.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+# Copyright 2020 The KubeCarrier Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu -o pipefail
+
+# Set OS specific values.
+if [[ "$OSTYPE" == "linux-gnu" ]]; then
+    BASE64_DECODE_FLAG="-d"
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+    BASE64_DECODE_FLAG="-D"
+elif [[ "$OSTYPE" == "linux-musl" ]]; then
+    BASE64_DECODE_FLAG="-d"
+else
+    echo "Unknown OS ${OSTYPE}"
+    exit 1
+fi
+
+kubectl apply -f - <<EOF >> /dev/null
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: kubecarrier-system
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: kubecarrier-sa
+  namespace: kubecarrier-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: kubecarrier-role
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - namespaces
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - get
+  - list
+  - update
+  - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: kubecarrier-crb
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: kubecarrier-role
+subjects:
+- kind: ServiceAccount
+  name: kubecarrier-sa
+  namespace: kubecarrier-system
+EOF
+# Get the service account token and CA cert.
+SA_SECRET_NAME=$(kubectl get -n kubecarrier-system sa/kubecarrier-sa -o "jsonpath={.secrets[0]..name}")
+# Note: service account token is stored base64-encoded in the secret but must
+# be plaintext in kubeconfig.
+SA_TOKEN=$(kubectl get -n kubecarrier-system secrets/${SA_SECRET_NAME} -o "jsonpath={.data['token']}" | base64 ${BASE64_DECODE_FLAG})
+CA_CERT=$(kubectl get -n kubecarrier-system secrets/${SA_SECRET_NAME} -o "jsonpath={.data['ca\.crt']}")
+
+# Extract cluster IP from the current context
+CURRENT_CONTEXT=$(kubectl config current-context)
+CURRENT_CLUSTER=$(kubectl config view -o jsonpath="{.contexts[?(@.name == \"${CURRENT_CONTEXT}\"})].context.cluster}")
+CURRENT_CLUSTER_ADDR=$(kubectl config view -o jsonpath="{.clusters[?(@.name == \"${CURRENT_CLUSTER}\"})].cluster.server}")
+
+echo "apiVersion: v1
```

Comment:
Maybe save kubeconfig to the `$1` and not echoing it to the stdout?

Reply:
I want to make it work the same as `kind get kubeconfig --internal --name eu-west-1 > /tmp/eu-west-1-kubeconfig`

Reply:
Hmmm why? In the end the kubeconfig is only useful if it's written to the file. Not sure can anybody consume it via stdin.

Reply:
`get-servicecluster-kubeconfig.sh > /tmp/eu-west-1-kubeconfig` same behaviour.
in other case, a user has to check script source code to find the name of the file